087-external-certificate-manager.md

```diff
@@ -0,0 +1,324 @@
+Strimzi will request the certificate in both PEM and PKCS12 format.
+Strimzi will wait for the usual operation timeout during the reconciliation loop for the `Certificate` status to indicate that the certificate has been issued before continuing.
+
+#### Tracking changes to KafkaUser certificates
+
+Cert-manager will be responsible for renewing all user certificates.
+When a certificate is renewed cert-manager will update the related Secret.
+Since the user's clients are directly using the cert-manager created Secret, Strimzi will take no action.
+
+#### Handling Clients CA cert renewals and key replacements
+
+During a reconciliation Strimzi will check the hash of the certificate stored in the user's CA public cert Secret to see if an update is needed.
+If the certificate has changed Strimzi will copy over the new certificate to replace the existing one.
+It will also update the `strimzi.io/ca-cert-hash` and increment the `strimzi.io/clients-ca-cert-generation` annotations.
+
+Once the annotations are updated Strimzi will update the annotation on the Kafka brokers Secret and the Kafka pods and roll the Kafka pods to trust the new CA cert.
+
+> ![Renewing/replacing the clients CA public cert/private key](./images/087-existing-renew-replace-clientca-certs.png)
+>
+> Fig 4: Existing workflow when the user provides a new clients CA public cert
+
+> ![Renewing/replacing the clients CA public cert/private key](./images/087-new-renew-replace-clientsca-cert.png)
+>
+> Fig 5: Proposed workflow when the user provides a new clients CA public cert
+
+## Affected/not affected projects
+
+This affects the Cluster Operator and User Operator.
+
+## Compatibility
+
+This feature will be optional and disabled by default.
+
+### Migrating to this feature
+
+To start using this feature in an existing Kafka cluster the user must:
+1. Install cert-manager and create an `Issuer`.
+2. Create Secrets to store the Cluster CA and/or Clients CA public certs
+3. Update the `Kafka` resource to have `clusterCa.type` and/or `clientsCa.type` set to `cert-manager.io`, `clusterCa.generateCertificateAuthority` and/or `clientsCa.generateCertificateAuthority` set to `false`, and `certManagerIssuerRef` and `publicCert` configured.
+
+When using this feature for the ClientsCa:
+* On the next Cluster operator reconciliation Strimzi will copy the new certificate over to `<CLUSTER_NAME>-clients-ca-cert` replacing the old certificate and increment the cert generation annotation.
+* Strimzi will roll the Kafka pods to trust the new CA cert.
+* On the next User Operator reconciliation Strimzi will create `Certificate` resources for all the existing `KafkaUser` custom resources.
+
+When using this feature for the ClusterCa:
+* On the next Cluster operator reconciliation Strimzi will copy the new certificate over to `<CLUSTER_NAME>-cluster-ca-cert` (renaming and keeping the old certificate), increment the cert generation annotation and add the key generation annotation.
```

Comment:
can we use the exact names of the annotations to be consistent across the rest of the proposal please? I think it's clearer.